hw2/01-coins.js

```diff
@@ -1,15 +1,63 @@
-/** Exercise 01 - Coins **/
+/* eslint-disable no-console */
+/** Exercise 01 - Coins * */
 
 const calculateChange = (input) => {
-  // Add your code here
-};
+    let result = `$${input} ==> `
+
+    if (input > 10.0 || input < 0.0) {
+        if (input > 10.0) result = result.concat('Error: the number is too large')
+        if (input < 0.0) result = result.concat('Error: the number is too small')
+        return result
+    }
+
+    let total = input * 100
+    const denom = [1, 5, 10, 25, 100]
+    const denomText = ['penn', 'nickel', 'dime', 'quarter', 'dollar']
+    let leading = false
+
+    while (total > 0) {
+        const coin = denom.pop()
+        let coinText = denomText.pop()
+
+        const amount = Math.floor(total / coin)
+        total -= amount * coin
+
+        // we have a vaid amount for the current denomination
+        if (amount > 0) {
+            // plurals
+            if (amount > 1) {
+                if (coinText === 'penn') {
+                    coinText = coinText.concat('ies') // pennies
+                } else {
+                    coinText = coinText.concat('s') // nickels, dimes, etc...
+                }
+            } else if (coinText === 'penn') {
+                coinText = coinText.concat('y') // penny
+            }
+            result =
+                leading === true
+                    ? result.concat(', ', amount, ' ', coinText)
+                    : result.concat(amount, ' ', coinText)
+
+            leading = true
+        }
+    }
+
+    return result
+}
```

Comment:
nice, clean code, handled all cases correctly. i also tested the function and it does what the test cases indicate it should.